fsharp-backend/src/BackendOnlyStdLib/LibDarkInternal.fs

```diff
@@ -765,6 +765,112 @@ human-readable data."
       deprecated = NotDeprecated }
 
 
+    { name = fn "DarkInternal" "startStaticAssetDeploy" 0
+      parameters = [ Param.make "username" TStr ""; Param.make "canvasID" TUuid "" ]
+      returnType =
+        TResult(
+          TRecord [ "deployHash", TStr
+                    "url", TStr
+                    "status", TStr
+                    "lastUpdate", TDate ],
+          TStr
+        )
+      description = "Records an in-progress static asset deployment"
+      fn =
+        internalFn (function
+          | _, [ DStr username; DUuid canvasID ] ->
+            uply {
+              match! Account.getUser (UserName.create username) with
+              | None ->
+                Exception.raiseInternal $"User not found" [ "username", username ]
+                return DResult(Error(DStr "User not found"))
+              | Some user ->
+                let! canvasMeta = Canvas.getMetaFromID canvasID
+                let! deploy =
+                  StaticAssets.startStaticAssetDeploy user canvasID canvasMeta.name
+                return
+                  DObj(
+                    Map [ "deployHash", DStr deploy.deployHash
+                          "url", DStr deploy.url
+                          "status", DStr(string deploy.status)
+                          "lastUpdate",
+                          DDate(DDateTime.fromInstant deploy.lastUpdate) ]
+                  )
+                  |> Ok
+                  |> DResult
+            }
+          | _ -> incorrectArgs ())
+      sqlSpec = NotQueryable
+      previewable = Impure
+      deprecated = NotDeprecated }
+
+
+    // TODO should this take username as a parameter?
+    // should user B be able to finish user A's upload?
```

Comment:
Outstanding question - if user A starts a static asset deploy, should other users be able to "finish" that deploy?
I don't really see any risk personally.